migration/idempotent/001-base.sql

```diff
@@ -2958,49 +2941,141 @@ BEGIN
             WHERE lkp.metric_name = m.metric_name
             ORDER BY key) label_keys,
         _prom_catalog.get_metric_retention_period(m.table_schema, m.metric_name) as retention_period,
```

Comment:
good idea. yeah, i'll try that

Reply:
I switched it to `COALESCE(m.retention_period, _prom_catalog.get_default_retention_period())` and it shaved 2 seconds off the 4.5 second runtime in my test system 🔥🔥🔥

Reply:
That made it 4.6 seconds. I believe that because `get_default_retention_period` is stable and takes no arguments, postgres is smart enough to run it once and cache the result. But if you materialize it in a CTE, then you have to cross join to the CTE to get it in the projection, and that slows it back down.

Reply:
Interesting. It made a substantial positive difference [here](https://github.com/timescale/promscale/pull/1634/files#diff-874005fbcf998a08680ed2ad95d2423302afa11ce0d7f68c0b6a7dee8a1e949cR108)

Reply:
I guess using it in a WHERE clause is not the same

Reply:
Or, perhaps, COALESCE doesn't evaluate it's second argument 🤔

Reply:
Ah, yeah, I bet it's the short-circuit logic in COALESCE